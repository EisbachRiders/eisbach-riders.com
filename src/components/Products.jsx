import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Img from 'gatsby-image'

import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'

import ProductDialog from './ProductDialog'

const styles = theme => ({
  root: {
    display: 'flex',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
  },
  container: {
    flexBasis: '30%',
  },
  img: {
    width: '100%',
  },
  name: {
    paddingTop: 15,
    paddingBottom: 10,
    fontWeight: 600,
    textAlign: 'left',
  },
  price: {
    color: theme.palette.primary.main,
    fontWeight: 600,
    display: 'flex',
    justifyContent: 'space-between',
    paddingBottom: 10,
  },
})

class Products extends Component {
  render() {
    const { images, classes } = this.props
    const products = [
      {
        name: 'Surfboard Thruster Fin Set - FCS Double Tab with Fin Key',
        images: [images[0], images[0], images[0], images[0], images[0]],
        price: '19.90',
        ebay: 'http://www.ebay.de/itm/183498144801',
        amazon: 'https://www.amazon.de/dp/B07K8W2TKK',
        more: [
          'Set of 3 fins with center fin and the two side fins (right and left) with inside profile in standard size G5 / M5.',
          'Fits any surfboard with double tab FCS1 plug system - whether bodyboard, shortboard, funboard or kiteboard, making it ideal for surfing, river surfing, paddle boarding or water sports in general.',
          'High quality materials make the fins durable and flexible.',
          'All our products come in a sustainable and environmentally friendly packaging.',
          "Height: 11.63 cm (4.58'')",
          "Length: 11.13 cm (4.38'')",
          "Thickness: 0.66 cm (0.26'')",
          'Angle: 35.5 degree',
          'Scope of delivery: 3 fin set with fin key',
        ],
      },
      {
        name: 'Surfboard RailProtect - Rail Protection Saver Tape',
        images: [images[2], images[2], images[2], images[2], images[2]],
        price: '24.90',
        ebay: 'http://www.ebay.de/itm/183498197638',
        amazon: 'https://www.amazon.de/dp/B07K8W9TG9',
        more: [
          'The particularly durable diamond structure protects the edges of your board against dings and bumps, thus preventing the ingress of water.',
          'Fits any surfboard up to 6ft (1.83m) - whether its bodyboard, shortboard, funboard or kiteboard, making it ideal for surfing, river surfing, paddle boarding or SUP (stand up paddling).',
          "The two adhesive strips (each 191 x 6.3 cm (6'3 x 2.5)) are easy to attach and will last a long time on your board.",
          'All our products come in a sustainable and environmentally friendly handcrafted packaging.',
          'Scope of delivery: 2 piece set surfboard RailProtect tape',
        ],
      },
      {
        name: 'Surfboard Thruster Fin Set - Future Single Tab with Fin Key',
        images: [images[0], images[0], images[0], images[0], images[0]],
        price: '19.90',
        ebay: 'http://www.ebay.de/itm/183498170584',
        amazon: null,
        more: [
          'Set of 3 fins with center fin and the two side fins (right and left) with inside profile in standard size G5 / M5.',
          'Fits any surfboard with single tab Future plug system - whether bodyboard, shortboard, funboard or kiteboard, making it ideal for surfing, river surfing, paddle boarding or water sports in general.',
          'High quality materials make the fins durable and flexible.',
          'All our products come in a sustainable and environmentally friendly packaging.',
          "Height: 11.63 cm (4.58'')",
          "Length: 11.13 cm (4.38'')",
          "Thickness: 0.66 cm (0.26'')",
          'Angle: 35.5 degree',
          'Scope of delivery: 3 fin set with fin key',
        ],
      },
    ]
    return (
      <div className={classes.root}>
        {products.map((elem, idx) => (
          <div className={classes.container} key={`product${idx}`}>
            <Img
              fluid={elem.images[0].node.fluid}
              alt=""
              className={classes.img}
            />
            <Typography className={classes.name}>{elem.name}</Typography>
            <Typography variant="h6" className={classes.price}>
              {`â‚¬${elem.price}`}
              <div>
                <Button style={{ width: '20%' }}>ebay</Button>
                <Button style={{ width: '20%' }}>a</Button>
              </div>
            </Typography>
            <ProductDialog product={elem} />
          </div>
        ))}
      </div>
    )
  }
}

Products.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Products)
