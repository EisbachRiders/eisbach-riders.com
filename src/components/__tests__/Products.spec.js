import React from 'react'
import { render } from 'react-testing-library'
import Products from '../Products'

describe('Products', () => {
  it('renders correctly', () => {
    const component = render(<Products />)
    expect(component).toMatchSnapshot()
  })
})
