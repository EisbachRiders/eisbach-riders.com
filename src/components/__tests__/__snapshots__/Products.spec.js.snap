// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hook-root-1inkmd6"
        id="products"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 Hook-title-1wwjqyh"
        >
          products.sectionTitle
        </h5>
        <div
          class="Hook-container-1nicdiq"
        >
          <div
            class="Hook-itemContainer-q53i6a"
          >
            <img
              alt="product 0"
              class="Hook-img-iydbqq"
              src="test-file-stub"
            />
            <div
              class="Hook-descriptionContainer-1p0knpg"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 Hook-name-1y8w25n"
              >
                FCS Double Tab Thruster
              </p>
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Hook-button-ot4ilw"
                href="https://amzn.to/2OfsJjc"
                rel="noopener"
                role="button"
                tabindex="0"
                target="_blank"
              >
                <span
                  class="MuiButton-label"
                >
                  common.more
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
          <div
            class="Hook-itemContainer-q53i6a"
          >
            <img
              alt="product 1"
              class="Hook-img-iydbqq"
              src="test-file-stub"
            />
            <div
              class="Hook-descriptionContainer-1p0knpg"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 Hook-name-1y8w25n"
              >
                RailProtect Tape
              </p>
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Hook-button-ot4ilw"
                href="https://amzn.to/2Y7BDDQ"
                rel="noopener"
                role="button"
                tabindex="0"
                target="_blank"
              >
                <span
                  class="MuiButton-label"
                >
                  common.more
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
          <div
            class="Hook-itemContainer-q53i6a"
          >
            <img
              alt="product 2"
              class="Hook-img-iydbqq"
              src="test-file-stub"
            />
            <div
              class="Hook-descriptionContainer-1p0knpg"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 Hook-name-1y8w25n"
              >
                Future Single Tab Thruster
              </p>
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Hook-button-ot4ilw"
                href="https://amzn.to/2OkhqGA"
                rel="noopener"
                role="button"
                tabindex="0"
                target="_blank"
              >
                <span
                  class="MuiButton-label"
                >
                  common.more
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hook-root-1inkmd6"
      id="products"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 Hook-title-1wwjqyh"
      >
        products.sectionTitle
      </h5>
      <div
        class="Hook-container-1nicdiq"
      >
        <div
          class="Hook-itemContainer-q53i6a"
        >
          <img
            alt="product 0"
            class="Hook-img-iydbqq"
            src="test-file-stub"
          />
          <div
            class="Hook-descriptionContainer-1p0knpg"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 Hook-name-1y8w25n"
            >
              FCS Double Tab Thruster
            </p>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Hook-button-ot4ilw"
              href="https://amzn.to/2OfsJjc"
              rel="noopener"
              role="button"
              tabindex="0"
              target="_blank"
            >
              <span
                class="MuiButton-label"
              >
                common.more
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="Hook-itemContainer-q53i6a"
        >
          <img
            alt="product 1"
            class="Hook-img-iydbqq"
            src="test-file-stub"
          />
          <div
            class="Hook-descriptionContainer-1p0knpg"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 Hook-name-1y8w25n"
            >
              RailProtect Tape
            </p>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Hook-button-ot4ilw"
              href="https://amzn.to/2Y7BDDQ"
              rel="noopener"
              role="button"
              tabindex="0"
              target="_blank"
            >
              <span
                class="MuiButton-label"
              >
                common.more
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="Hook-itemContainer-q53i6a"
        >
          <img
            alt="product 2"
            class="Hook-img-iydbqq"
            src="test-file-stub"
          />
          <div
            class="Hook-descriptionContainer-1p0knpg"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 Hook-name-1y8w25n"
            >
              Future Single Tab Thruster
            </p>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Hook-button-ot4ilw"
              href="https://amzn.to/2OkhqGA"
              rel="noopener"
              role="button"
              tabindex="0"
              target="_blank"
            >
              <span
                class="MuiButton-label"
              >
                common.more
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
